tools/Dashboard/DevHome.Dashboard/Controls/WidgetControl.xaml.cs

```diff
@@ -47,7 +47,7 @@ public WidgetControl()
         ActualThemeChanged += OnActualThemeChanged;
     }
 
-    private void OpenWidgetMenu(object sender, RoutedEventArgs e)
+    private async void OpenWidgetMenuAsync(object sender, RoutedEventArgs e)

```

Comment:
Should this account for concurrent execution when the button is clicked multiple times?

Reply:
It's not exactly scientific, but I've never gotten this to do anything weird. I think because the next click will be dismissing the menu, you can't really get it to run concurrently.